brax/training/agents/ppo/networks_vision.py

```diff
@@ -0,0 +1,149 @@
+"""PPO vision networks."""
+
+from typing import Any, Callable, Mapping, Sequence, Tuple, Union
+
+import jax
+import jax.numpy as jp
+import flax
+from flax import linen
+from flax.core import FrozenDict
+
+from brax.training import distribution
+from brax.training import networks
+from brax.training import types
+from brax.training.types import PRNGKey
+from brax.training.agents.ppo.networks_cnn import VisionMLP
+
+
+ModuleDef = Any
+ActivationFn = Callable[[jp.ndarray], jp.ndarray]
+Initializer = Callable[..., Any]
+
+
+@flax.struct.dataclass
+class PPONetworks:
+  policy_network: networks.FeedForwardNetwork
+  value_network: networks.FeedForwardNetwork
+  parametric_action_distribution: distribution.ParametricDistribution
+
+
+def remove_pixels(obs: Union[jp.ndarray, Mapping]) -> Union[jp.ndarray, Mapping]:
+  """Remove pixel observations from the observation dict.
+  FrozenDicts are used to avoid incorrect gradients."""
+  if not isinstance(obs, Mapping):
+    return obs
+  obs = FrozenDict(obs)
+  pixel_keys = [k for k in obs.keys() if k.startswith("pixels/")]
+  state_obs = obs
+  for k in pixel_keys:
+    state_obs, _ = state_obs.pop(k)
+  return state_obs
+
+
+def make_policy_network_vision(
+  observation_size: Mapping[str, Tuple[int, ...]],
+  output_size: int,
+  preprocess_observations_fn: types.PreprocessObservationFn = types.identity_observation_preprocessor,
+  hidden_layer_sizes: Sequence[int] = [256, 256],
+  activation: ActivationFn = linen.swish,
+  kernel_init: Initializer = jax.nn.initializers.lecun_uniform(),
+  layer_norm: bool = False,
+  state_obs_key: str = "",
+  normalise_channels: bool = False,
+) -> networks.FeedForwardNetwork:
+  module = VisionMLP(
+    layer_sizes=list(hidden_layer_sizes) + [output_size],
+    activation=activation,
+    kernel_init=kernel_init,
+    layer_norm=layer_norm,
+    normalise_channels=normalise_channels,
+    state_obs_key=state_obs_key,
+  )
+
+  def apply(processor_params, policy_params, obs):
+    obs = FrozenDict(obs)
+    if state_obs_key:
+      state_obs = preprocess_observations_fn(remove_pixels(obs), processor_params)
```

Comment:
i'm not quite sure i follow this.  it seems like you are doing this:

1) preprocess all non-pixel fields (i assume this is for normalization)
2) take only a SINGLE field (the state obs key) from the result and pass it into the module

if that's the case, why not just preprocess the state obs key?